### PR TITLE
Adding ability to add Suppresions of metapackages that are tfm-specific

### PR DESCRIPTION
**Pull request by joperezr**
cc: @ericstj @safern @ViktorHofer 

This plus it's corresponding ingestion into dotnet/runtime will fix https://github.com/dotnet/runtime/issues/39641

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.Packaging/src/ApplyMetaPackages.cs [hunk 1]
```diff
@@ -56,13 +57,35 @@ public override bool Execute()
             var index = PackageIndex.Load(PackageIndexes.Select(pi => pi.GetMetadata("FullPath")));
             List<ITaskItem> updatedDependencies = new List<ITaskItem>();
 
-            var suppressMetaPackages = new HashSet<string>(SuppressMetaPackages.NullAsEmpty(), StringComparer.OrdinalIgnoreCase);
+            var suppressMetaPackages = new Dictionary<string, HashSet<string>>();
+
+            if (SuppressMetaPackages != null)
+            {
+                foreach (var metapackage in SuppressMetaPackages)
```
Comment (safern) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/src/ApplyMetaPackages.cs:64]:
```suggestion
                foreach (string metapackage in SuppressMetaPackages)
```

File: src/Microsoft.DotNet.Build.Tasks.Packaging/src/ApplyMetaPackages.cs [hunk 2]
```diff
@@ -56,13 +57,35 @@ public override bool Execute()
...
+                    if (string.IsNullOrEmpty(tfmSpecificSupression))
+                    {
+                        // If the supression doesn't specify a TargetFramework, then it applies to all.
+                        suppressMetaPackages[metapackage.ItemSpec].Add("All");
+                    }
+                    else
+                    {
+                        var fx = NuGetFramework.Parse(tfmSpecificSupression);
+                        suppressMetaPackages[metapackage.ItemSpec].Add(fx.DotNetFrameworkName);
+                    }
```
Comment (safern) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/src/ApplyMetaPackages.cs:80]: We could simplify this:

```suggestion
                    if (!suppressMetaPackages.TryGetValue(metapackage.ItemSpec, out var value))
                    {
                        value = new HashSet<string>();
                        suppressMetaPackages.Add(metapackage.ItemSpec, value);
                    }
                    var tfmSpecificSupression = metapackage.GetMetadata("TargetFramework");
                    if (string.IsNullOrEmpty(tfmSpecificSupression))
                    {
                        // If the supression doesn't specify a TargetFramework, then it applies to all.
                        value.Add("All");
                    }
                    else
                    {
                        var fx = NuGetFramework.Parse(tfmSpecificSupression);
                        value.Add(fx.DotNetFrameworkName);
                    }
```

File: src/Microsoft.DotNet.Build.Tasks.Packaging/src/ApplyMetaPackages.cs [hunk 3]
```diff
@@ -100,6 +123,16 @@ public override bool Execute()
...
         }
 
+        private bool ShouldSuppressMetapackage(Dictionary<string, HashSet<string>> suppressedMetaPackages, string metaPackage, NuGetFramework tfm)
+        {
+            if (suppressedMetaPackages.ContainsKey(metaPackage) &&
+                (suppressedMetaPackages[metaPackage].Contains("All") || suppressedMetaPackages[metaPackage].Contains(tfm.DotNetFrameworkName)))
+            {
+                return true;
+            }
+            return false;
```
Comment (safern) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/src/ApplyMetaPackages.cs:133]:
```suggestion
            return suppressedMetaPackages.TryGetValue(metaPackage, out var value) &&
                (value.Contains("All") || value.Contains(tfm.DotNetFrameworkName))
```

**Comment by joperezr** [comment]
I locally built dotnet/runtime latest using these changes, and then diffed the .nuspecs of all of our packages just for sanity check and did see that the only differences are with packages:
```
Microsoft.Extensions.DependencyModel
Microsoft.Extensions.FileProviders.Physical
Microsoft.Extensions.Hosting
Microsoft.Extensions.Logging.Console
```
All of which were previously having a dependency on NETStandard.Library package in .NET Framework 4.6.1 and now have a dependency to whichever particular package was promoting this instead.

**Pull request merged**